server/duplicate-handler.js

```diff
@@ -0,0 +1,90 @@
+const debug = require('debug')('duplicateHandler');
+
+function duplicateHandler({ generateRequestIdentifier, skip, timeout } = {}) {
+  timeout = Number(timeout) || 30;
+
+  const requests = {};
+
+  // Garbage collection (not necessary under normal operation)
+  const gc = () => {
+    const ids = Object.keys(requests);
+    if (ids.length > 0) {
+      debug(`${ids.length} current registered requests`);
+    }
+    for (const id of ids) {
+      if (requests[id].registeredAt < new Date().getTime() - timeout * 1000) {
+        delete requests[id];
+      }
+    }
+  };
+
+  setInterval(gc, 1000);
+
+  return function handleDuplicate(req, res, next) {
+    if (skip && skip(req)) {
+      next();
+      return;
+    }
+
+    const id = generateRequestIdentifier ? generateRequestIdentifier(req) : req.url;
```

Comment:
I don't see a `generateRequestIdentifier` passed as param, so `req.url` will be used as an id each time right? If that's the case, we should make sure there's no login token or API key passed to not serve any private info